### ISSUE TITLE
uBoost Convergence

### ISSUE THREAD

**Issue by gmarceca** [arogozhnikov/hep_ml#52]
Hello,

How could I check the convergence of uBoost when using uniforming_rate != 0?. When I plot the log-loss metric vs number of boostings I see it increases, with a rate proportional to the uniforming_rate value used. You can see this trend in the plot attached (for privacy regulations I had to remove the legend and so it affected the bottom right plot, but you can appreciate the effect anyway). On the other hand, I can make the log-loss to decrease with another hyper-parameter configuration but then I don't get an uniform selection. How can I deal with this?, does it mean that the log-loss is not a good metric to check convergence in this case?. 

[plot_uboost_vs_adaboost.pdf](https://github.com/arogozhnikov/hep_ml/files/2102588/plot_uboost_vs_adaboost.pdf)
Thanks very much,
Gino

**Comment by arogozhnikov** [comment]
Hi Gino, 

for uBoost convergence is something poorly defined. 

- first, uBoost has no optimization target (contrary say to AdaBoost, GBDT, GB+FL)   
- second, the way it operates quite often drives to bias in weights in one of direction - thus probabilities may become very biased, thus loss may diverge

Among options, I think you can monitor ROC AUC on validation set or some similar discriminative measure.
